nixos/modules/services/web-apps/kanboard.nix

```diff
@@ -0,0 +1,158 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.kanboard;
+
+  toStringAttrs = lib.mapAttrs (lib.const toString);
+  mkDefaultAttrs = lib.mapAttrs (_: lib.mkDefault);
+in
+{
+  meta.maintainers = with lib.maintainers; [ yzx9 ];
+
+  options.services.kanboard = {
+    enable = lib.mkEnableOption "Kanboard";
+
+    package = lib.mkPackageOption pkgs "kanboard" { };
+
+    dataDir = lib.mkOption {
+      type = lib.types.str;
+      default = "/var/lib/kanboard";
+      description = "Default data folder for Kanboard.";
+      example = "/mnt/kanboard";
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "kanboard";
+      description = "User under which Kanboard runs.";
+    };
+
+    settings = lib.mkOption {
+      type =
+        with lib.types;
+        attrsOf (oneOf [
+          str
+          int
+          bool
+        ]);
+
+      default = { };
+
+      description = ''
+        Customize the default settings, refer to <https://github.com/kanboard/kanboard/blob/main/config.default.php>
+        for details on supported values.
+      '';
+    };
+
+    # Nginx
+    domain = lib.mkOption {
+      type = lib.types.str;
+      default = "kanboard";
+      description = "FQDN for the Kanboard instance.";
+      example = "kanboard.example.org";
+    };
+    nginx = lib.mkOption {
+      type = lib.types.nullOr (
+        lib.types.submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; })
+      );
+      default = { };
+      description = ''
+        With this option, you can customize an NGINX virtual host which already
+        has sensible defaults for Kanboard. Set to `{ }` if you do not need any
+        customization for the virtual host. If enabled, then by default, the
+        {option}`serverName` is `''${domain}`. If this is set to null (the
+        default), no NGINX virtual host will be configured.
+      '';
+      example = lib.literalExpression ''
+        {
+          enableACME = true;
+          forceHttps = true;
+        }
+      '';
+    };
+
+    phpfpm.settings = lib.mkOption {
+      type =
+        with lib.types;
+        attrsOf (oneOf [
+          int
+          str
+          bool
+        ]);
+      default = mkDefaultAttrs {
+        "pm" = "dynamic";
+        "php_admin_value[error_log]" = "stderr";
+        "php_admin_flag[log_errors]" = true;
+        "listen.owner" = "nginx";
+        "catch_workers_output" = true;
+        "pm.max_children" = "32";
+        "pm.start_servers" = "2";
+        "pm.min_spare_servers" = "2";
+        "pm.max_spare_servers" = "4";
+        "pm.max_requests" = "500";
+      };
+
+      description = ''
+        Options for kanboard's PHPFPM pool.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      group = config.services.nginx.group;
```

Comment:
Is there a specific reason this needs to run under the NGINX group by the way? Usually services will be able to run under a different user/group than that of the web server

Reply:
There’s no specific reason for this. I noticed that other `nixos/web-apps` mix different groups, with some, like [castopod](https://github.com/NixOS/nixpkgs/blob/6d8ddd87dc098ac9fb0468a80d2d94c400c5f856/nixos/modules/services/web-apps/castopod.nix#L312-L316), [cloudlog](https://github.com/NixOS/nixpkgs/blob/6d8ddd87dc098ac9fb0468a80d2d94c400c5f856/nixos/modules/services/web-apps/cloudlog.nix#L496-L500), [filesender](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-apps/filesender.nix), using the NGINX group, while others, like [Writefreely](https://github.com/NixOS/nixpkgs/blob/f49f947ec4a6d313697595e17ff06568c65abdab/nixos/modules/services/web-apps/writefreely.nix#L156-L160), use a different group. I also couldn’t find a official guideline covering this topic.

Although I don't think it's strictly required, I do agree with you that using a different group might be better. So, I’ll change it to an option.